packages/jest-core/src/runJest.ts

```diff
@@ -242,9 +242,22 @@ export default async function runJest({
   }
 
   if (changedFilesPromise) {
-    testSchedulerContext.changedFiles = (
-      await changedFilesPromise
-    ).changedFiles;
+    const changedFilesInfo = await changedFilesPromise;
+    if (changedFilesInfo.changedFiles) {
+      testSchedulerContext.changedFiles = changedFilesInfo.changedFiles;
+      const sourcesRelatedToTestsInChangedFilesArray = contexts
+        .map(context => {
+          const searchSource = new SearchSource(context);
```

Comment:
Creating the SearchSource for a given context is already done earlier, when finding related tests. We could optimize this by storing the `searchSources` in and array, when mapping `context`s for `testRunData` and reuse it. I think it's worth not doubling the efforts we already made.